modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -281,6 +297,16 @@
     private ConcurrentMap<UUID, TxTimeoutOnPartitionMapExchangeChangeFuture> txTimeoutOnPartitionMapExchangeFuts =
         new ConcurrentHashMap<>();
 
+    /** Timeout operations. */
+    private final Map<String, GridTimeoutProcessor.CancelableTask> TIMEOUT_OPS =
+        new HashMap<String, GridTimeoutProcessor.CancelableTask>() {{
+            put(IGNITE_LONG_OPERATIONS_DUMP_TIMEOUT, null);
+            put(IGNITE_DUMP_TX_COLLISIONS_INTERVAL, null);
+    }};
+
+    /** Key collisions info holder. */
+    private KeyCollisionsHolder keyCollisionsInfo;
```

Comment:
i make it volatile, think it would be enough, if we would have situation with initialized GridDistributedCacheEntry and still not initialized IgniteTxManager we are in big troubles i hope. For example check additionally : org.apache.ignite.internal.processors.cache.transactions.IgniteTxManager#deferredAckMsgSnd what do u think here ?